packages/portalnetwork/src/networks/history/history.ts

```diff
@@ -52,6 +52,19 @@ export class HistoryNetwork extends BaseNetwork {
    * @returns content if available locally
    */
   public findContentLocally = async (contentKey: Uint8Array): Promise<Uint8Array | undefined> => {
+    const contentType = contentKey[0]
+    if (contentType === HistoryNetworkContentType.BlockHeaderByNumber) {
+      const blockNumber = '0x' + decodeHistoryNetworkContentKey(contentKey).keyOpt.toString(16)
+      const index = await this.blockIndex()
```

Comment:
I can't remember why we did things this way before but why do we have to get a local reference to the block index?  And why is it on the base network class and not just history?  At a minimum, let's replace this sort of getter function and just expose the blockIndex publicly, or else write some useful wrappers around it for getting the block hash/number from it's mate.

Reply:
Not sure exactly, but I think because we imagined the blockIndex being updated / accessed by multiple networks
Regardless, the way we go about it can be nicer, for sure

Reply:
For now, I implemented the wrapper `this.blockNumberToHash`